src/amber/controller/render.cr

```diff
@@ -37,5 +37,43 @@ module Amber::Controller
         Kilt.render("#{{{path}}}/{{filename.id}}")
       {% end %}
     end
+
+    protected def respond_with(html : String? = nil, json : Hash | String? = nil, xml : String? = nil, text : String? = nil)
+      accepts = context.request.headers["Accept"].split(";").try(&.split(","))
```

Comment:
There should probably be a guard check for when Accept is `*/*`. also sometimes the accept header could be in this format`text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8` Also we have some helper methods that you can make use of in this case, see https://github.com/amberframework/amber/blob/master/src/amber/support/mime_types.cr

Reply:
@eliasjpr Thanks. I'm thinking that we should probably just default to html or text or something it doesn't match any of accept types. Seems like pretty much all browsers have */* as the last set of accepts parameters.

How would you suggest the mime_type helpers could be used: It seems like `Amber::Support::Mime.mime_type("json")` is more complicated than `"application/json"`. Since all of the ones we care about only point to one mime type I feel that it's probably bests just to hardcode the string. Thoughts?

Reply:
Well, the Amber::Support::Mime has a Hash of mime-types you can probably write a nice matcher method in that module to match the one from the header, this will keep the code centralized in one place instead of hardcoding the string.